### PR TITLE
Add Micronaut MyBatis guide

### PR DESCRIPTION
**Pull request by ZacharyKlein**

### CODE REVIEW

File: guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+include::{commondir}/common-create-app.adoc[]
+
+=== Configure Data Source and JPA
+
+Add the next snippet to `build.gradle` to include the necessary dependencies:
+
+:dependencies:
+
+dependency:mybatis:3.4.6[groupId=org.mybatis,callout=1]
```
Comment (ilopmar) [inline guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc]: Please make sure you use the latest MyBatis version: https://mvnrepository.com/artifact/org.mybatis/mybatis

File: guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+dependency:micronaut-flyway[groupId=io.micronaut.flyway]
+
+Configure the database migrations directory for Flyway in `application.yml`.
+
+resource:application.yml[tag=flyway]
+
+Create the file `V1__schema.sql` with the database schema creation:
+
+resource:databasemigrations/V1__schema.sql[]
```
Comment (ilopmar) [inline guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc]: Please, change this and the app code to use the default location for the migrations. We changed it sometime ago in Flyway integration so it's good to have this guide aligned. See https://micronaut-projects.github.io/micronaut-flyway/latest/guide/#jpa

File: guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+14:55:12.407 [Test worker] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
+14:55:12.692 [Test worker] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
+14:55:12.875 [Test worker] INFO  o.f.c.i.license.VersionPrinter - Flyway Community Edition 5.2.1 by Boxfuse
+14:55:12.915 [Test worker] INFO  o.f.c.i.database.DatabaseFactory - Database: jdbc:h2:mem:default (H2 1.4)
+14:55:13.052 [Test worker] INFO  o.f.core.internal.command.DbValidate - Successfully validated 1 migration (execution time 00:00.025s)
+14:55:13.071 [Test worker] INFO  o.f.c.i.s.JdbcTableSchemaHistory - Creating Schema History table: "PUBLIC"."flyway_schema_history"
+14:55:13.112 [Test worker] INFO  o.f.core.internal.command.DbMigrate - Current version of schema "PUBLIC": << Empty Schema >>
+14:55:13.118 [Test worker] INFO  o.f.core.internal.command.DbMigrate - Migrating schema "PUBLIC" to version 1 - schema
+14:55:13.157 [Test worker] INFO  o.f.core.internal.command.DbMigrate - Successfully applied 1 migration to schema "PUBLIC" (execution time 00:00.087s)
+----
```
Comment (ilopmar) [inline guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc]: This needs to be build tool agnostic. You can do something like:

```
Run the tests:

:exclude-for-build:maven

[source, bash]
----
$ ./gradlew test
----

:exclude-for-build:

:exclude-for-build:gradle

[source, bash]
----
$ ./mvnw test
----

:exclude-for-build:
```

File: guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+$ curl http://localhost:8080/genres/list
+[{"id":1,"name":"Sci-fi"},{"id":2,"name":"Science"}]
+
+$ curl -X DELETE http://localhost:8080/genres/1
+
+$ curl http://localhost:8080/genres/list
+[{"id":2,"name":"Science"}]
+----
+
```
Comment (ilopmar) [inline guides/micronaut-data-access-mybatis/micronaut-data-access-mybatis.adoc:199]: Add here the section for GraalVM (change the texts if you want):

```
include::{commondir}/common-graal-with-plugins.adoc[]

:exclude-for-languages:groovy

You can execute the same curl request as before to check that the native image works.

:exclude-for-languages:
```

**Review by ilopmar — CHANGES_REQUESTED** [review]
Besides the comments in the specific parts of the guide, you also need to:
- Remove [these lines](https://github.com/micronaut-projects/micronaut-guides-poc/blob/master/guides/micronaut-data-access-mybatis/metadata.json#L8-L9) to run the tests
- Create a `MyBatis` feature in https://github.com/micronaut-projects/micronaut-guides-poc/tree/master/buildSrc/src/main/java/io/micronaut/guides/feature and use it in https://github.com/micronaut-projects/micronaut-guides-poc/blob/master/guides/micronaut-data-access-mybatis/metadata.json#L13

Once you have it, build the guide again and make sure you run the generated `test.sh` script for the guide and all test pass.

**Comment by ZacharyKlein** [comment]
@ilopmar I believe I was able to resolve the last test failure (there was a change in MyBatis 3.5.x that required a tweak to the annotations for generated keys). This is ready for review again.

**Review by ilopmar — APPROVED** [review]
Excellent!

If just pushed a minor fix to make the guide build-tool agnostic and also to remove the lookup for a specific MyBatis version. It is pulled automatically from the version in pom.xml (that is update by dependabot).

**Pull request merged**